link/src/main/java/com/stripe/android/link/repositories/LinkApiRepository.kt

```diff
@@ -234,6 +235,23 @@ internal class LinkApiRepository @Inject constructor(
         )
     }
 
+    override suspend fun updatePaymentDetails(
+        updateParams: ConsumerPaymentDetailsUpdateParams,
+        consumerSessionClientSecret: String,
+        consumerPublishableKey: String?
+    ): Result<ConsumerPaymentDetails> {
+        return stripeRepository.updatePaymentDetails(
+            clientSecret = consumerSessionClientSecret,
+            paymentDetailsUpdateParams = updateParams,
+            requestOptions = consumerPublishableKey?.let {
+                ApiRequest.Options(it)
+            } ?: ApiRequest.Options(
+                publishableKeyProvider(),
+                stripeAccountIdProvider()
+            )
```

Comment:
I think we should really consider doing the suggestion I made in this comment: https://github.com/stripe/stripe-android/pull/9710#discussion_r1859448571

do you have any other new APIs planned for this project? If so, can you do that follow up before adding more?

Reply:
Yh, there's one more api call to add. I'll fix the issue before that.